src/core/credentials/transport/alts/grpc_alts_credentials_options.h

```diff
@@ -35,6 +39,7 @@ typedef struct grpc_alts_credentials_options_vtable {
 struct grpc_alts_credentials_options {
   const struct grpc_alts_credentials_options_vtable* vtable;
   grpc_gcp_rpc_protocol_versions rpc_versions;
+  std::optional<std::vector<std::string>> record_protocols;
```

Comment:
Putting a C++ type in this struct means that we have to switch to using C++ `new` and `delete` for allocation, instead of C-style `malloc()` and `free()`.  Otherwise, we will never actually call the ctor for this member, which is unsafe for a type that we don't own.

Reply:
I really don't think it's going to be a lot of work to change this to use `new` and `delete`.  There should only be a handful of lines of code that need to change here.